### PR TITLE
Add OPAI & support wan event

### PR DESCRIPTION
**Pull request by lwindg**

### CODE REVIEW

File: schema/index.yaml [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+        "source": "eth0",
+        "dns": [
+          "8.8.8.8",
+          "8.8.4.4",
+        ]
+      }
+    DNS_PUT:
+      {
+        "enableFixed": false,
+        "fixedDNS": [],
```
Comment (zack9433) [inline schema/index.yaml]: 是不是有少 `dns` 這個 key

Comment (lwindg) [inline schema/index.yaml]: put 不希望 user 改 fixed dns 以外的資訊。web 目前也沒提供吧？

File: bundle.json [hunk 2]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "dns",
-  "version": "0.10.3",
+  "version": "0.11.1",
```
Comment (imZack) [inline bundle.json]: +1 on Major number due to breaking changes

File: dns.py [hunk 3]
```diff
@@ -295,13 +295,23 @@ def _event_network_interface(self, message):
...
 
+    @Route(methods="put", resource="/network/wan")
+    def _event_network_wan(self, message):
+        """
+        Listen wan event to update the dns settings.
+        """
+        try:
+            self.set_current_dns({"source": message.data["interface"]})
+        except Exception as e:
+            _logger.debug("[/network/wan] %s".format(e.message))
```
Comment (imZack) [inline dns.py]: debug or info level?

File: schema/index.yaml [hunk 4]
```diff
@@ -0,0 +1,109 @@
...
+            {
+              "application/json": {
+                $ref: '#/externalDocs/x-mocks/DNS_PUT'
+              }
+            }
+definitions:
+  DNS_GET:
+    title: DNS_GET
+    required:
+      - enableFixed
```
Comment (imZack) [inline schema/index.yaml]: List with no indent

File: schema/index.yaml (hunk 1 above)
Comment (zack9433) [inline schema/index.yaml]: 那 bundle 要擋設定 `dns` property 唷

File: bundle.json (hunk 2 above)
Comment (lwindg) [inline bundle.json]: no breaking change, API is the same

File: schema/index.yaml (hunk 1 above)
Comment (lwindg) [inline schema/index.yaml]: Done

File: dns.py (hunk 3 above)
Comment (lwindg) [inline dns.py]: info, thank you.

**Pull request merged**